Add recipe for pomo-cat

kn66 opened the pull request:
### Brief summary of what the package does

pomo-cat.el is a simple yet charming Pomodoro timer for Emacs. Unlike many existing Pomodoro packages that rely on notifications, mode-line displays, or logging, pomo-cat places an adorable cat in the center of your screen during breaks—either as an image or ASCII art.

### Direct link to the package repository

https://github.com/kn66/pomo-cat.el

### Your association with the package

Author + Maintainer

### Relevant communications with the upstream package maintainer

[e.g., `package.el` compatibility changes that you have submitted. Write **None needed** if there is no problem.]

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass -

⸺ `pomo-cat.el` with byte-compile using Emacs 30.1:
```
pomo-cat.el:138:38: Warning: the function `posframe-poshandler-frame-center' might not be defined at runtime.
pomo-cat.el:135:10: Warning: the function `posframe-show' might not be defined at runtime.
pomo-cat.el:97:8: Warning: the function `posframe-delete' might not be defined at runtime.
pomo-cat.el:96:12: Warning: the function `posframe-workable-p' might not be defined at runtime.
```
^ these are because you're using `eval-when-compile` - don't use it in this case, these are functions not macros.

⸺ `pomo-cat.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- pomo-cat.el#L233: Only use back/front quotes to link to top-level elisp symbols
```

⸺ Package and license:
- pomo-cat.el needs *formal* license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)

kn66 commented:
Thank you for the quick review! I've addressed all the feedback:

  **Byte-compile warnings**: Removed `eval-when-compile` for posframe - now using regular `(require 'posframe
  nil t)` so the functions are available at runtime.

  **Backticks in docstring**: Fixed improper use of backticks around `SECONDS` parameter in the docstring.

  **License**: Added `SPDX-License-Identifier: MIT` to meet the license requirement.

  The package should now be ready for MELPA. Please let me know if there are any other issues!

riscy commented:
Thank you!  And welcome to MELPA. :)
